webapp/security/views.py

```diff
@@ -333,5 +342,185 @@ def cve(cve_id):
+            for rel in package_data["releases"]:
+                releases.append(
+                    PackageReleaseStatus(
+                        name=rel["name"],
+                        status=rel["status"],
+                        status_description=rel["status_description"],
+                    )
+                )
+            packages.append(
+                Package(
+                    name=package_data["name"],
+                    source=package_data["source"],
+                    ubuntu=package_data["ubuntu"],
+                    debian=package_data["debian"],
+                    releases=releases,
+                )
+            )
+
+    if "references" in data and len(data["references"]) > 0:
+        for ref in data["references"]:
+            references.append(CVEReference(uri=ref))
+
+    if "bugs" in data and len(data["bugs"]) > 0:
+        for b in data["bugs"]:
+            bugs.append(Bug(uri=b))
+
+    cves = [
+        CVE(
+            id=data["id"],
+            status=data.get("status", ""),
+            last_updated_date=data.get("last_updated_date"),
+            public_date_usn=data.get("public_date_usn"),
+            priority=data.get("priority"),
+            cvss=data.get("cvss"),
+            assigned_to=data.get("assigned_to"),
+            discovered_by=data.get("discovered_by"),
+            approved_by=data.get("approved_by"),
+            description=data.get("description"),
+            ubuntu_description=data.get("ubuntu_description"),
+            notes=data.get("notes", []),
+            packages=packages,
+            references=references,
+            bugs=bugs,
+        )
+    ]
+
+    try:
+        db_session.add_all(cves)
+        db_session.commit()
+    except exc.IntegrityError as error:
+        return flask.jsonify({"message": error.orig.args[0]}), 400
+    except exc.SQLAlchemyError as error:
+        return flask.jsonify({"message": error}), 400
+
+    return flask.jsonify({"message": "CVE created succesfully"}), 201
+
+
+def update_cve():
+    data = flask.request.json
+    cve = db_session.query(CVE).filter(CVE.id == data["id"]).first()
+
+    # Check if CVE exists by candidate
+    if len(db_session.query(CVE).all()) == 0:
+        response = flask.jsonify({"message": "CVE does not exist"}), 400
+        return response
+
+    cve.status = data.get("status")
+    cve.last_updated_date = data.get("last_updated_date")
+    cve.public_date_usn = data.get("public_date_usn")
+    cve.public_date = data.get("public_date")
+    cve.priority = data.get("priority")
+    cve.crd = data.get("crd")
+    cve.cvss = data.get("cvss")
+    cve.assigned_to = data.get("assigned_to")
+    cve.discovered_by = data.get("discovered_by")
+    cve.approved_by = data.get("approved_by")
+    cve.description = data.get("description")
+    cve.ubuntu_description = data.get("ubuntu_description")
+    cve.notes = data.get("notes")
+
+    if "references" in data and len(data["references"]) > 0:
```

Comment:
Assuming `data["references"]` will contain a list, you don't need to test `len(data["references"]) > 0` because `bool([]) == False`. An empty list is Falsey.